frontend/src/lib/components/LemonInput/LemonInput.tsx

```diff
@@ -12,11 +12,12 @@ export interface LemonInputProps
     > {
     ref?: React.Ref<HTMLInputElement>
     id?: string
+    type?: string
     value?: string | number
     defaultValue?: string
     placeholder?: string
-    onChange?: (newValue: string) => void
-    onPressEnter?: (newValue: string) => void
+    onChange?: (newValue: string | number) => void
+    onPressEnter?: (newValue: string | number) => void
```

Comment:
This isn't very ergonomic at the moment, and makes the typing check fail because existing code is incompatible – TypeScript should be informed that `type="string"` (or unspecified `type`) components can only handle strings and `type="string"` only numbers. Type inference based on interface unions can be the solution. Actually, I gave this a go in #10545 to see if this is workable so that I don't lead you down a blind alley 😅 but fortunately it's all good.

Reply:
Merged in, thanks again!